include/onnxruntime/core/common/common.h

```diff
@@ -95,6 +99,57 @@ void LogRuntimeError(uint32_t session_id, const common::Status& status, const ch
 
+#ifdef ORT_NO_EXCEPTIONS
+
+#define ORT_TRY
+
+// Throw an exception with optional message.
+// NOTE: The arguments get streamed into a string via ostringstream::operator<<
+// DO NOT use a printf format string, as that will not work as you expect.
+#define ORT_THROW(...)                                                                       \
+  do {                                                                                       \
+    std::cerr << ::onnxruntime::OnnxRuntimeException(ORT_WHERE_WITH_STACK,                   \
+                                                     ::onnxruntime::MakeString(__VA_ARGS__)) \
+                     .what()                                                                 \
+              << std::endl;                                                                  \
+    abort();                                                                                 \
+  } while (false)
+
+// Just in order to mark things as not implemented. Do not use in final code.
+#define ORT_NOT_IMPLEMENTED(...)                                                                \
+  do {                                                                                          \
+    std::cerr << ::onnxruntime::NotImplementedException(::onnxruntime::MakeString(__VA_ARGS__)) \
+                     .what()                                                                    \
+              << std::endl;                                                                     \
+    abort();                                                                                    \
+  } while (false)
+
+// Check condition.
+// NOTE: The arguments get streamed into a string via ostringstream::operator<<
+// DO NOT use a printf format string, as that will not work as you expect.
+#define ORT_ENFORCE(condition, ...)                                                            \
+  do {                                                                                         \
+    if (!(condition)) {                                                                        \
+      std::cerr << ::onnxruntime::OnnxRuntimeException(ORT_WHERE_WITH_STACK, #condition,       \
+                                                       ::onnxruntime::MakeString(__VA_ARGS__)) \
+                       .what()                                                                 \
+                << std::endl;                                                                  \
+      abort();                                                                                 \
+    }                                                                                          \
+  } while (false)
+
+#define ORT_THROW_EX(ex)           \
+  do {                             \
+    std::cerr << #ex << std::endl; \
+    abort();                       \
+  } while (false)
+
+#define ORT_THROW_BAD_ALLOC ORT_THROW_EX(std::bad_alloc)
+
```

Comment:
Could possibly be a bit fancier here and provide a facility to throw an arbitrary exception with 0 or 1 arguments. Adapted from [https://stackoverflow.com/a/8814003/684911](https://stackoverflow.com/a/8814003/684911)

Following is what you'd have when exceptions are enabled. Adjust accordingly for when they're disabled.
```
#define ORT_THROW_EX_0ARGS(ex) throw ex()
#define ORT_THROW_EX_1ARG(ex, A) throw ex(A)

// The interim macro that simply strips the excess and ends up with the required macro
#define ORT_THROW_EX_ARG_MATCHER(ex, noargs_placeholder, A, FUNC, ...) FUNC

// The macro that the programmer uses
#define ORT_THROW_EX(ex, ...) ORT_THROW_EX_ARG_MATCHER(ex, , ##__VA_ARGS__,                \
                                                       ORT_THROW_EX_1ARG(ex, __VA_ARGS__), \
                                                       ORT_THROW_EX_0ARGS(ex, __VA_ARGS__))
```